wordpress-mobile/WordPress-Android
# Issue/13455 make post as seen

Fixes #13455

This PR adds an ability to mark posts as seen or unseen manually, or marking posts as seen by opening them, similarly to Calypso.

This functionality is available to all users logged in with wpcom account for posts that are a part of a feed (search or tag (topic) posts are excluded) . Logged-out users will not be able to access the toggle button in post card menu, and opening a post will not mark as seen, since there is no access token.

@develric I used your implementation of Comment Following as a reference, so the code should hopefully be familiar :)

[![Image from Gyazo](https://i.gyazo.com/6b1a395981a31404437467bf21695361.gif)](https://gyazo.com/6b1a395981a31404437467bf21695361)


To test:
- Enable `UNREAD_POSTS_COUNT` feature.
- Open Reader and navigate to any stream (for test purposes navigate to the stream with a Site Filter)
- Open site filter and take a note of the unread post count (will not be visible if all posts are red).
- Open postcard menu and mark post as seen or unseen. Notice the Snackbar confirming the success of operation. Open the Reader in Calypso and confirm that seen status of the post was changed.
- Open site filter and confirm that the unread post counter was incremented or decremented.
- Turn off internet connection and try toggling seen status of the post. Notice Snackbar with error message.

PR submission checklist:
